### ISSUE TITLE
Keycloak in restart loop after upgrade to 24.0.4

### ISSUE THREAD

**Issue by pureiboi** [keycloak/keycloak#30129]
### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

core

### Describe the bug

I tried to upgrade from version 22.0.0 to 24.0.4, it went into restart loop.  and there is no clue on on this upgrade.

P.S I have successfully upgraded in the lower environment for the same configuration.

logs that I managed to capture

[keycloak-5cbd586d85-7mrlf2.log](https://github.com/user-attachments/files/15545666/keycloak-5cbd586d85-7mrlf2.log)
[keycloak-5cbd586d85-7mrlf-1.log](https://github.com/user-attachments/files/15545668/keycloak-5cbd586d85-7mrlf-1.log)

### Version

24.0.4

### Regression

- [ ] The issue is a regression

### Expected behavior

service is up and running

### Actual behavior

restart loop

### How to Reproduce?

upgrade from version 22.0.0 to 24.0.4

### Anything else?

i observed the following exception, not sure how relevant it is to the issue

2024-06-04 08:04:19,526 DEBUG [org.hibernate.orm.model.mapping.creation] (JPA Startup Thread) Mapping-model creation encountered (possibly) transient error : java.lang.IllegalStateException: Not yet ready

**Comment by brentpi** [comment]
@pureiboi

I had a similar issue and was scratching my head looking for the problem - and had the same error message as yours (but, it's not fatal).

My problem ended up being a misconfiguration.  With v22, if you go to the http root; you get the old keycloak index page (status code 200); but in v24 that has changed to a redirect (302) to /admin/master/console/.

So, if you have a health check looking for 200 on / - ECS or whatever you're using to run keycloak may constantly try to restart the container.

Hope this helps!

**Comment by pureiboi** [comment]
thanks @brentpi 

it's funny, the behavior is not consistent.  

keycloak is running on k8s,  running on 3 environment. i.e both **dev** and **uat** are working fine, only **staging** is having issues.  all are using the same config and probe endpoint i.e `/auth/realms/master`

I have switched to health check endpoints i.e `/auth/health/live`,  **staging** is still a headache 

---

oh yeah, I forgot to highlight the liquibase migration failure. There were multiple different kinds of errors on md5sum, exceptions. 

I wonder if there is any way to rebuild the liquibase databaselogs? 
I have been trying to update the records, copying the records from another env into **staging**. 

```
Picked up JAVA_TOOL_OPTIONS:  -javaagent:/otel-auto-instrumentation-java/javaagent.jar
OpenJDK 64-Bit Server VM warning: Sharing is only supported for boot loader classes because bootstrap classpath has been appended
[otel.javaagent 2024-06-07 15:56:32:421 +0000] [main] INFO io.opentelemetry.javaagent.tooling.VersionLogger - opentelemetry-javaagent - version: 1.32.0
2024-06-07 15:56:47,225 ERROR [liquibase.changelog.ChangeSet] (main) ChangeSet META-INF/jpa-changelog-23.0.0.xml::23.0.0-17258::keycloak encountered an exception.
UPDATE SUMMARY
Run:                          8
Previously run:             116
Filtered out:                 0
-------------------------------
Total change sets:          124
2024-06-07 15:56:47,323 ERROR [org.keycloak.quarkus.runtime.cli.ExecutionExceptionHandler] (main) ERROR: Failed to start server in (production) mode
2024-06-07 15:56:47,324 ERROR [org.keycloak.quarkus.runtime.cli.ExecutionExceptionHandler] (main) ERROR: Failed to update database
2024-06-07 15:56:47,324 ERROR [org.keycloak.quarkus.runtime.cli.ExecutionExceptionHandler] (main) ERROR: liquibase.exception.LiquibaseException: liquibase.exception.MigrationFailedException: Migration failed for changeset META-INF/jpa-changelog-23.0.0.xml::23.0.0-17258::keycloak:
     Reason: liquibase.exception.DatabaseException: ERROR: column "details_json_long_value" of relation "event_entity" already exists [Failed SQL: (0) ALTER TABLE keycloak.EVENT_ENTITY ADD DETAILS_JSON_LONG_VALUE TEXT]
2024-06-07 15:56:47,324 ERROR [org.keycloak.quarkus.runtime.cli.ExecutionExceptionHandler] (main) ERROR: liquibase.exception.MigrationFailedException: Migration failed for changeset META-INF/jpa-changelog-23.0.0.xml::23.0.0-17258::keycloak:
     Reason: liquibase.exception.DatabaseException: ERROR: column "details_json_long_value" of relation "event_entity" already exists [Failed SQL: (0) ALTER TABLE keycloak.EVENT_ENTITY ADD DETAILS_JSON_LONG_VALUE TEXT]
2024-06-07 15:56:47,324 ERROR [org.keycloak.quarkus.runtime.cli.ExecutionExceptionHandler] (main) ERROR: Migration failed for changeset META-INF/jpa-changelog-23.0.0.xml::23.0.0-17258::keycloak:
     Reason: liquibase.exception.DatabaseException: ERROR: column "details_json_long_value" of relation "event_entity" already exists [Failed SQL: (0) ALTER TABLE keycloak.EVENT_ENTITY ADD DETAILS_JSON_LONG_VALUE TEXT]
2024-06-07 15:56:47,325 ERROR [org.keycloak.quarkus.runtime.cli.ExecutionExceptionHandler] (main) ERROR: ERROR: column "details_json_long_value" of relation "event_entity" already exists [Failed SQL: (0) ALTER TABLE keycloak.EVENT_ENTITY ADD DETAILS_JSON_LONG_VALUE TEXT]
2024-06-07 15:56:47,325 ERROR [org.keycloak.quarkus.runtime.cli.ExecutionExceptionHandler] (main) ERROR: ERROR: column "details_json_long_value" of relation "event_entity" already exists
2024-06-07 15:56:47,325 ERROR [org.keycloak.quarkus.runtime.cli.ExecutionExceptionHandler] (main) For more details run the same command passing the '--verbose' option. Also you can use '--help' to see the details about the usage of the particular command.
```

**Comment by pureiboi** [comment]
putting notes here in case others faced similar issues as mine

I'm running containerized keycloak on k8s cluster, on AWS fargate sitting within appmesh.

A few things I have noted during the debugging process related to restart loop related to liveness probe
- liquibase migration - resolved with update md5sum or insert missing records due to constraints
- db connection exception (no impact though, since all firewalls are cleared )
- update to health check endpoints - from /auth/realms/master to /auth/health/ready

all above didn't help

---

So I went further to disable liveness probe, and discovered that the service is up and running, however the access from both ALB/NLB or port-forward still doesn't work

Given that I got a different error from curl, I got HTTP 503 from the response. I realized there was mtls configured, which differs from my other env. 

and that finally solved the mystery.
